Makefile

```diff
@@ -13,7 +13,7 @@ BUILD_PROTO=BUILD_PROTO=1
 
 .PHONY: clippy
 clippy: ## Runs Clippy with configs
-	cargo clippy --locked --workspace --all-targets --all-features -- -D warnings
+	cargo clippy --locked --workspace --all-targets --all-features -- -D warnings --allow clippy::arc_with_non_send_sync
```

Comment:
This seems really unsafe 😅 But I don't understand these unsafe impl context's so I'll withhold judgement until its been explained.

I would suggest manually `#[allow(arc_with_non_send_sync)]` on the actual instances instead of blanket allowing this.

Reply:
No, it's safe, it's just a suppression of annoying clippy warnings introduced in the latest Rust versions. :) It warns you, if you use `Arc` where it is sufficient to use just `Rc`. I think, we should update our code and switch this lint on, but in separate PR.

Reply:
Let's create an issue to remove this once https://github.com/0xPolygonMiden/miden-base/issues/909 is fixed.